discovery/aws/ec2.go

```diff
@@ -124,17 +125,28 @@ func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return err
 	}
 	if c.Region == "" {
-		sess, err := session.NewSession()
+		// TODO(@sysadmind): Should we get a context from somewhere?
+		ctx := context.TODO()
+		cfg, err := awsConfig.LoadDefaultConfig(ctx)
 		if err != nil {
 			return err
 		}
-		metadata := ec2metadata.New(sess)
-		region, err := metadata.Region()
-		if err != nil {
-			return errors.New("EC2 SD configuration requires a region")
+
+		if cfg.Region != "" {
+			// If the region is already set in the config, use it.
+			// This can happen if the user has set the region in the AWS config file or environment variables.
+			c.Region = cfg.Region
+		} else {
+			// Attempt to get the region from the instance metadata service.
+			metaClient := imds.NewFromConfig(cfg)
```

Comment:
Looking from the code I wonder if LoadDefaultConfig does not do that by default.

See https://github.com/aws/aws-sdk-go-v2/blob/1eea3dbe07dffcef2fdba6f8e2206b32b91d8a83/config/config.go#L58

Reply:
I think you're correct. https://github.com/aws/aws-sdk-go-v2/blob/main/config/resolve.go#L383